scripts/sign_encrypt.py

```diff
@@ -29,391 +59,876 @@ def int_parse(str):
+            offs += len(self.attr)
+            self.name_img = self.inf[offs:offs + self.name_size]
+            offs += self.name_size
+            self.next_inf = self.inf[offs:]
+
+            def find_attr(attr):
+                if self.attr_count <= 0:
+                    return None
+                for n in range(self.attr_count):
+                    o = subkey_offs + UUID_SIZE + SK_HDR_SIZE + n * 12
+                    [attr_value, attr_offs,
+                     attr_len] = struct.unpack('<III', self.inf[o: o + 12])
+                    if attr_value == attr:
+                        o = subkey_offs + attr_offs
+                        return self.inf[o:o + attr_len]
+                return None
+
+
+            n_bytes = find_attr(TEE_ATTR_RSA_MODULUS)
+            e_bytes = find_attr(TEE_ATTR_RSA_PUBLIC_EXPONENT)
+            e = int.from_bytes(e_bytes, 'big')
+            n = int.from_bytes(n_bytes, 'big')
+            self.subkey_key = rsa.RSAPublicNumbers(e, n).public_key()
+
+            self.img = self.inf[subkey_offs:offs - self.name_size]
+            if len(self.img) != img_size:
+                raise Exception("Unexpected img size: got {}, expected {}"
+                                .format(len(self.img), img_size))
+        else:
+            raise Exception("Unsupported image type: {}".format(img_type))
+
+
+    def display(self):
+        import binascii
+        import struct
+        import uuid
+
+        def display_ta():
+            nonlocal offs
+            ta_uuid = self.inf[offs:offs + UUID_SIZE]
+            print('  uuid:       {}'.format(uuid.UUID(bytes=ta_uuid)))
+            offs += UUID_SIZE
+            [ta_version] = struct.unpack('<I', self.inf[offs:offs + 4])
+            print('  ta_version: {}'.format(ta_version))
+
+            offs += 4;
+            if img_type == SHDR_ENCRYPTED_TA:
+                ehdr = self.inf[offs: offs + EHDR_SIZE]
+                offs += EHDR_SIZE
+                [enc_algo, flags, nonce_len,
+                 tag_len] = struct.unpack('<IIHH', ehdr)
+
+                enc_algo_name = 'Unkown'
+                if enc_algo in enc_tee_alg.values():
+                    enc_algo_name = value_to_key(enc_tee_algo, enc_algo)
+                print('  enc_algo:   {} ({})'.format(enc_algo, enc_algo_name))
+
+                if enc_algo not in enc_tee_alg.values():
+                    raise Exception('Unrecognized encrypt algorithm: 0x{:08x}'
+                                    .format(enc_value))
+                print('  nonce_len:  {} (bytes)'.format(nonce_len))
+                if nonce_len != NONCE_SIZE:
+                    raise Exception("Unexpected nonce len: {}"
+                                    .format(nonce_len))
+                nonce = self.inf[offs:offs + nonce_len]
+                print('  nonce:      {}...'.format(binascii.hexlify(nonce)))
+                offs += nonce_len
+
+                print('  tag_len:    {} bytes'.format(tag_len))
```

Comment:
It would be better to use the exact same name as in the C structure: `iv_size`, `tag_size`. Maybe print/decode flags too while we're at it?